openreview/conference/templates/programchairWebfield.js

```diff
@@ -254,6 +281,102 @@ var displayHeader = function() {
   Webfield.ui.tabPanel(tabs);
 };
 
+var displayConfiguration = function(requestForm, invitations) {
+
+  var formatPeriod = function(invitation) {
+    var start;
+    var end;
+    var afterStart = true;
+    var beforeEnd = true;
+    var now = Date.now();
+    if (invitation.cdate) {
+      var date = new Date(invitation.cdate);
+      start =  date.toLocaleDateString('en-GB', { hour: 'numeric', minute: 'numeric', day: '2-digit', month: 'short', year: 'numeric', timeZoneName: 'long'});
+      afterStart = now > invitation.cdate;
+    }
+    if (invitation.duedate) {
+      var date = new Date(invitation.duedate);
+      end =  date.toLocaleDateString('en-GB', { hour: 'numeric', minute: 'numeric', day: '2-digit', month: 'short', year: 'numeric', timeZoneName: 'long'});
+      beforeEnd = now < invitation.duedate;
+    }
+
+    var currentPeriod = afterStart && beforeEnd;
+    var periodString = start ? 'from <i>' + start + '</i> ' : 'open ';
+    if (end) {
+      periodString = periodString + 'until <i>' + end + '</i>';
+    } else {
+      periodString = periodString + 'no deadline';
+    }
+
+    return periodString;
+  }
+  var invitaitonMap = {};
+
+  invitations.forEach(function(invitation) {
+    invitaitonMap[invitation.id] = invitation;
+  });
+
+  var container = '#venue-configuration';
+  var html = '<div id="container"></br>'
+
+  if (requestForm) {
+    html = html +  '<a href="/forum?id=' + requestForm.id + '">Requested configuration</a><span>: ' + getConfigurationDescription(requestForm) + '</span></br></br>';
+  }
+
+  html = html + '<span>Official Committee: </span>' +
+    '<a href="/group?id=' + PROGRAM_CHAIRS_ID + '&mode=edit">Program Chairs</a>, ';
+
+  if (SHOW_AC_TAB) {
+    html = html + '<a href="/group?id=' + AREA_CHAIRS_ID + '&mode=edit">Area Chairs</a> (' +
+      '<a href="/group?id=' + AREA_CHAIRS_ID + '/Invited&mode=edit">Invited</a>, ' +
+      '<a href="/group?id=' + AREA_CHAIRS_ID + '/Declined&mode=edit">Declined</a>), ';
+  }
+
+  html = html + '<a href="/group?id=' + REVIEWERS_ID + '&mode=edit">Reviewers</a> (' +
+    '<a href="/group?id=' + REVIEWERS_ID + '/Invited&mode=edit">Invited</a>, ' +
+    '<a href="/group?id=' + REVIEWERS_ID + '/Declined&mode=edit">Declined</a>), ' +
+    '<a href="/group?id=' + AUTHORS_ID + '&mode=edit">Authors</a></br></br>' +
+    '<h3>Timeline:</h3></br>' +
+    '<ul>';
+
+  var invitation = invitaitonMap[SUBMISSION_ID];
+  if (invitation) {
+    html = html + '<li><a href="/invitation?id=' + invitation.id + '">Paper Submission</a> ' + formatPeriod(invitation) + '</li>';
+  };
+
+  var invitation = invitaitonMap[CONFERENCE_ID + '/-/' + BID_NAME];
```

Comment:
Doesn't this produce a keyError if the Bid invitation is missing?

Reply:
no, in javascript you get an 'undefined'. There is an If after this line